pages/design_documentation/ocf_io_engines.md

```diff
@@ -0,0 +1,289 @@
+where the missing data will eventually be inserted into the cache to accelerate 
+subsequent reads.
+
+In case of a partial miss and if some target sectors are dirty (partial dirty miss),
+cleaning is performed first and then the data will be read from the core.
+
+
+## Write Back (WB)
+The write-back engine writes the data to a cache device only. The data will be 
+stored on a core device only after being cleaned - flushed to core device.
+This engine is used primarily in the Write Back cache mode to accelerate both 
+write and read operations at the cost of a higher risk of data loss if the cache 
+drive fails. 
+
+See [**write back cache mode configuration**](/cache_configuration.html#write-back)
+for details.
+	
+Write Back cache mode defines its read and write operations as follows:
+~~~~
+[OCF_IO_WB_IF] = {
+        .read = ocf_read_generic,
+        .write = ocf_write_wb,
+        .name = "Write Back"
+},      
+~~~~
+
+### Write Back - write path
+The requested data is written to the cache device only, producing dirty data.
+This data is then periodically cleaned by the [**cleaner**](/cleaning.html). 
+
+The request may be handled by [**Pass Through**](/ocf_io_engines.html#pass-through-pt) 
+engine if it cannot be handled by Write Back method. 
+See [**Write Through write path**](/ocf_io_engines.html#write-through---write-path).
+
+### Write Back - read path
+Read requests are handled by the generic read function, as in 
+[**Write Through read path**](/ocf_io_engines.html#write-through---read-path).
+
+
+## Write Around (WA)
+This engine combines the logic of Write Through and Write Invalidate engines.
+The data will land on core device. It may be stored in the cache only if there 
+is already a valid mapping.
+
+Write Around cache mode is used for disjointed workloads, where read requests 
+are accelerated with cache, but written data are effectively filtered out of 
+cache. See [**write around cache mode configuration**](/cache_configuration.html#write-around)  
+It uses the following functions:
+~~~~
+[OCF_IO_WA_IF] = {
+        .read = ocf_read_generic,
+        .write = ocf_write_wa,
+        .name = "Write Around"
+},
+~~~~
+
+### Write Around - write path
+This engine writes the data to the cache and to the core (by executing 
+[**Write Through write**](/ocf_io_engines.html#write-through---write-path)
+internally) **if and only if the data has already been mapped**. Otherwise, 
+the data goes to the core device only and are invalidated in the cache (by executing 
+[**Write Invalidate write**](/ocf_io_engines.html#write-invalidate---write-path) 
+internally).
+
+### Write Around - read path
+Read requests are handled by the generic read function, as in 
+[**Write Through read path**](/ocf_io_engines.html#write-through---read-path).
+
+
+## Write Invalidate (WI)
+This engine is accessible only via OpenCAS Framework API as there is no use case
+for it in OpenCAS Linux. 
```

Comment:
I don't know if this commend is needed here. Btw most of the engines are hidden from OCL.

Reply:
I commented it in this way because one cannot use this interface/cache mode in OpenCAS Linux, yet it's available through OCF API